Rewrite DiscoveryRule API test to pytest

dvagala opened the pull request:
Test results:
```
(robotello-master) [dvagala@fedora robottelo]$ pytest tests/foreman/api/test_discoveryrule.py 
================================================ test session starts ================================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/dvagala/rh/robottelo
plugins: forked-1.1.3, mock-1.10.4, services-1.3.1, cov-2.8.1, xdist-1.31.0
collecting ... 2020-04-28 09:56:49 - conftest - DEBUG - Collected 2 test cases
2020-04-28 11:56:51 - robottelo.ssh - DEBUG - Instantiated Paramiko client id
2020-04-28 11:56:51 - robottelo.ssh - INFO - Connected to [FQDN]
2020-04-28 11:56:51 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-04-28 11:56:53 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-7.el7sat.noarch

2020-04-28 11:56:53 - robottelo.ssh - DEBUG - Destroyed Paramiko client id
2020-04-28 11:56:53 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-04-28 11:56:53 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 2 items                                                                                                   

tests/foreman/api/test_discoveryrule.py ..                                                                    [100%]

============================================= 2 passed in 69.98 seconds =============================================

rplevka commented on tests/foreman/api/test_discoveryrule.py:
```diff
@@ -24,91 +24,95 @@
+
+
+@tier1
+def test_negative_create_with_invalid_host_limit_and_priority():
```
the tested attributes could be parametrized here, but i think we're good for now.
The whole test should be probably moved upstream, outside of robottelo (to the rails integration tests) - that should be the plan for the future.

dvagala commented:
> Just out of curiosity. How does the log look like? (generated robottelo.log).
> or could you run this with `-s` option (so pytest doesn't supress the stdout)?
> I'm trying to see whether we didn't lose any of the logging capabilities.
> 
> Otherwise ACK

It basically missing just those  "Started setUpClass" and "Started tearDownClass" logs, and less http communication logs, because I've dropped some testing.
Now as I see it here, I've noticed as I've moved creating and removing entities to fixtures, those logs are now before "Started Test" and "Finished Test" logs. That would be a problem right?


**This is current log output (the lines are cuted):**
```
(robotello-master) [dvagala@fedora robottelo]$ pytest tests/foreman/api/test_discoveryrule.py -s
============================= test session starts ==============================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/dvagala/rh/robottelo
plugins: forked-1.1.3, mock-1.10.4, services-1.3.1, cov-2.8.1, xdist-1.31.0
2020-05-11 08:56:02 - conftest - DEBUG - Collected 2 test cases
2020-05-11 10:56:03 - robottelo.ssh - DEBUG - Instantiated Paramiko client id
2020-05-11 10:56:03 - robottelo.ssh - INFO - Connected to [FQDN]
2020-05-11 10:56:03 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-05-11 10:56:05 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-7.el7sat.noarch

2020-05-11 10:56:05 - robottelo.ssh - DEBUG - Destroyed Paramiko client id
2020-05-11 10:56:05 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-05-11 10:56:05 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 2 items

tests/foreman/api/test_discoveryrule.py 2020-05-11 10:56:05 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/katello/api/v2/organizations with options {'auth': (''
2020-05-11 10:56:11 - nailgun.client - DEBUG - Received HTTP 200 response:   {"label":"SsetRBR","owner_details":{"created":"2020-05-11T08:56:09+0000","updated":"2020-05-11T08:56:09+0

2020-05-11 10:56:11 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/katello/api/v2/organizations/364 with options {'auth': ('', ''), 'verify': False, 'headers': {'
2020-05-11 10:56:12 - nailgun.client - DEBUG - Received HTTP 200 response: {"label":"SsetRBR","owner_details":{"created":"2020-05-11T08:56:09+0000","updated":"2020-05-11T08:56:09+000
2020-05-11 10:56:12 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/locations with options {'auth': ('', ''), 'verify': False, 'headers': {'content-type': 
2020-05-11 10:56:16 - nailgun.client - DEBUG - Received HTTP 201 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"created_at":"2020-05-11 08:56:
2020-05-11 10:56:16 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/locations/365 with options {'auth': ('', ''), 'verify': False, 'headers': {'content-type
2020-05-11 10:56:17 - nailgun.client - DEBUG - Received HTTP 200 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"created_at":"2020-05-11 08:56:
2020-05-11 10:56:17 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/hostgroups with options {'auth': ('', ''), 'verify': False, 'headers': {'content-type':
2020-05-11 10:56:19 - nailgun.client - DEBUG - Received HTTP 201 response: {"subnet_id":null,"subnet_name":null,"operatingsystem_id":null,"operatingsystem_name":null,"domain_id":null
2020-05-11 10:56:19 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/hostgroups/265 with options {'auth': ('', ''), 'verify': False, 'headers': {'content-typ
2020-05-11 10:56:20 - nailgun.client - DEBUG - Received HTTP 200 response: {"content_source_id":null,"content_source_name":null,"content_view_id":null,"content_view_name":null,"lifec
2020-05-11 10:56:20 - robottelo - DEBUG - Started Test: test_discoveryrule.py/test_positive_end_to_end_crud
2020-05-11 10:56:20 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/discovery_rules with options {'auth': ('', ''), 'verify': False, 'headers': {'content-t
2020-05-11 10:56:21 - nailgun.client - DEBUG - Received HTTP 201 response: {"id":192,"name":"㤧䗜酪𗰟绱筦徾랏ꫢ䄸㟛鮐玢痺𐠏ﶽ𗬦圾성ꩢ짛ﭞ𞸒𓅛ᘼ윂纹殜槟扔鷃慦ᬙ뜸昒䰃䩗𘆲掊㑉歋쓟𘓞篬兙玱㝠㙅粭Ţ屍ꕦ㵀餠傮𘏟콂ᐡꊧㄴ봜𓄔㩕ꋡⱍ렛ម䛲ﲇ앃","search"
2020-05-11 10:56:21 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/discovery_rules/192 with options {'auth': ('', ''), 'verify': False, 'headers': {'conten
2020-05-11 10:56:21 - nailgun.client - DEBUG - Received HTTP 200 response: {"name":"㤧䗜酪𗰟绱筦徾랏ꫢ䄸㟛鮐玢痺𐠏ﶽ𗬦圾성ꩢ짛ﭞ𞸒𓅛ᘼ윂纹殜槟扔鷃慦ᬙ뜸昒䰃䩗𘆲掊㑉歋쓟𘓞篬兙玱㝠㙅粭Ţ屍ꕦ㵀餠傮𘏟콂ᐡꊧㄴ봜𓄔㩕ꋡⱍ렛ម䛲ﲇ앃","enabled":true,"h
2020-05-11 10:56:21 - nailgun.client - DEBUG - Making HTTP PUT request to https://FQDN/api/v2/discovery_rules/192 with options {'auth': ('', ''), 'verify': False, 'headers': {'conten
2020-05-11 10:56:23 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":192,"name":"㤧䗜酪𗰟绱筦徾랏ꫢ䄸㟛鮐玢痺𐠏ﶽ𗬦圾성ꩢ짛ﭞ𞸒𓅛ᘼ윂纹殜槟扔鷃慦ᬙ뜸昒䰃䩗𘆲掊㑉歋쓟𘓞篬兙玱㝠㙅粭Ţ屍ꕦ㵀餠傮𘏟콂ᐡꊧㄴ봜𓄔㩕ꋡⱍ렛ម䛲ﲇ앃","search"
2020-05-11 10:56:23 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/katello/api/v2/organizations/364 with options {'auth': ('', ''), 'verify': False, 'headers': {'
2020-05-11 10:56:24 - nailgun.client - DEBUG - Received HTTP 200 response: {"label":"SsetRBR","owner_details":{"created":"2020-05-11T08:56:09+0000","updated":"2020-05-11T08:56:09+000
2020-05-11 10:56:24 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/locations/365 with options {'auth': ('', ''), 'verify': False, 'headers': {'content-type
2020-05-11 10:56:25 - nailgun.client - DEBUG - Received HTTP 200 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"created_at":"2020-05-11 08:56:
2020-05-11 10:56:25 - nailgun.client - DEBUG - Making HTTP PUT request to https://FQDN/api/v2/discovery_rules/192 with options {'auth': ('', ''), 'verify': False, 'headers': {'conten
2020-05-11 10:56:27 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":192,"name":"卉戨扱𒁃覸橅ಡ붞㩷甿㬣𐲚덗䶢𒁁㬤峸𑀄ꚃ㗽켩𗛋亩蓹ੜ긵ࢣ𘌖퇠물됺𗵓𛄓蚖𗆺든𗔶잟𗡰𘄾謶鞙𗝳枚ⲑ𗚑溈","search":"Location = Defau
2020-05-11 10:56:27 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/discovery_rules/192 with options {'auth': ('', ''), 'verify': False, 'headers': {'conten
2020-05-11 10:56:27 - nailgun.client - DEBUG - Received HTTP 200 response: {"name":"卉戨扱𒁃覸橅ಡ붞㩷甿㬣𐲚덗䶢𒁁㬤峸𑀄ꚃ㗽켩𗛋亩蓹ੜ긵ࢣ𘌖퇠물됺𗵓𛄓蚖𗆺든𗔶잟𗡰𘄾謶鞙𗝳枚ⲑ𗚑溈","enabled":false,"hostgroup_id":265,
2020-05-11 10:56:27 - nailgun.client - DEBUG - Making HTTP PUT request to https://FQDN/api/v2/discovery_rules/192 with options {'auth': ('', ''), 'verify': False, 'headers': {'conten
2020-05-11 10:56:28 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":192,"name":"卉戨扱𒁃覸橅ಡ붞㩷甿㬣𐲚덗䶢𒁁㬤峸𑀄ꚃ㗽켩𗛋亩蓹ੜ긵ࢣ𘌖퇠물됺𗵓𛄓蚖𗆺든𗔶잟𗡰𘄾謶鞙𗝳枚ⲑ𗚑溈","search":"Location = Defau
2020-05-11 10:56:28 - nailgun.client - DEBUG - Making HTTP DELETE request to https://FQDN/api/v2/discovery_rules/192 with options {'auth': ('', ''), 'verify': False, 'headers': {'con
2020-05-11 10:56:29 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":192,"name":"卉戨扱𒁃覸橅ಡ붞㩷甿㬣𐲚덗䶢𒁁㬤峸𑀄ꚃ㗽켩𗛋亩蓹ੜ긵ࢣ𘌖퇠물됺𗵓𛄓蚖𗆺든𗔶잟𗡰𘄾謶鞙𗝳枚ⲑ𗚑溈","search":"Location = Defau
2020-05-11 10:56:29 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/discovery_rules/192 with options {'auth': ('', ''), 'verify': False, 'headers': {'conten
2020-05-11 10:56:31 - nailgun.client - WARNING - Received HTTP 404 response: {
  "error": {"message":"Resource discovery_rule not found by id '192'"}
}

.2020-05-11 10:56:31 - robottelo - DEBUG - Finished Test: test_discoveryrule.py/test_positive_end_to_end_crud
2020-05-11 10:56:31 - robottelo - DEBUG - Started Test: test_discoveryrule.py/test_negative_create_with_invalid_host_limit_and_priority
2020-05-11 10:56:31 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/hostgroups with options {'auth': ('', ''), 'verify': False, 'headers': {'content-type':
2020-05-11 10:56:31 - nailgun.client - DEBUG - Received HTTP 201 response: {"subnet_id":null,"subnet_name":null,"operatingsystem_id":null,"operatingsystem_name":null,"domain_id":null
2020-05-11 10:56:31 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/hostgroups/266 with options {'auth': ('', ''), 'verify': False, 'headers': {'content-typ
2020-05-11 10:56:32 - nailgun.client - DEBUG - Received HTTP 200 response: {"content_source_id":null,"content_source_name":null,"content_view_id":null,"content_view_name":null,"lifec
2020-05-11 10:56:32 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/discovery_rules with options {'auth': ('', ''), 'verify': False, 'headers': {'content-t
2020-05-11 10:56:33 - nailgun.client - WARNING - Received HTTP 422 response: {
  "error": {"id":null,"errors":{"max_count":["is not a number"]},"full_messages":["Max count is not a number"]}
}

2020-05-11 10:56:33 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/hostgroups with options {'auth': ('', ''), 'verify': False, 'headers': {'content-type':
2020-05-11 10:56:35 - nailgun.client - DEBUG - Received HTTP 201 response: {"subnet_id":null,"subnet_name":null,"operatingsystem_id":null,"operatingsystem_name":null,"domain_id":null
2020-05-11 10:56:35 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/hostgroups/267 with options {'auth': ('', ''), 'verify': False, 'headers': {'content-typ
2020-05-11 10:56:36 - nailgun.client - DEBUG - Received HTTP 200 response: {"content_source_id":null,"content_source_name":null,"content_view_id":null,"content_view_name":null,"lifec
2020-05-11 10:56:36 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/discovery_rules with options {'auth': ('', ''), 'verify': False, 'headers': {'content-t
2020-05-11 10:56:37 - nailgun.client - WARNING - Received HTTP 422 response: {
  "error": {"id":null,"errors":{"priority":["is not a number"]},"full_messages":["Priority is not a number"]}
}

.2020-05-11 10:56:37 - robottelo - DEBUG - Finished Test: test_discoveryrule.py/test_negative_create_with_invalid_host_limit_and_priority
2020-05-11 10:56:37 - nailgun.client - DEBUG - Making HTTP DELETE request to https://FQDN/api/v2/hostgroups/265 with options {'auth': ('', ''), 'verify': False, 'headers': {'content-
2020-05-11 10:56:38 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":265,"name":"rSSaURyGmX","created_at":"2020-05-11T08:56:18.997Z","updated_at":"2020-05-11T08:56:18.997
2020-05-11 10:56:38 - nailgun.client - DEBUG - Making HTTP DELETE request to https://FQDN/api/v2/locations/365 with options {'auth': ('', ''), 'verify': False, 'headers': {'content-t
2020-05-11 10:56:39 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":365,"name":"rXYcKUoQGsRN","created_at":"2020-05-11T08:56:13.509Z","updated_at":"2020-05-11T08:56:13.5
2020-05-11 10:56:39 - nailgun.client - DEBUG - Making HTTP DELETE request to https://FQDN/katello/api/v2/organizations/364 with options {'auth': ('', ''), 'verify': False, 'headers':
2020-05-11 10:56:41 - nailgun.client - DEBUG - Received HTTP 202 response:   {"id":"cf20f75f-6d77-4bcd-85d9-567c84bb2910","label":"Actions::Katello::Organization::Destroy","pending":

2020-05-11 10:56:41 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/foreman_tasks/api/tasks/cf20f75f-6d77-4bcd-85d9-567c84bb2910 with options {'auth': ('', ''), 'v
2020-05-11 10:56:43 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":"cf20f75f-6d77-4bcd-85d9-567c84bb2910","label":"Actions::Katello::Organization::Destroy","pending":fa


========================== 2 passed in 40.82 seconds ===========================


```








**This is former log output without this PR:**
```
(robotello-master) [dvagala@fedora robottelo]$ pytest tests/foreman/api/test_discoveryrule.py -s
============================= test session starts ==============================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/dvagala/rh/robottelo
plugins: forked-1.1.3, mock-1.10.4, services-1.3.1, cov-2.8.1, xdist-1.31.0
2020-05-11 09:04:39 - conftest - DEBUG - Collected 2 test cases
2020-05-11 11:04:40 - robottelo.ssh - DEBUG - Instantiated Paramiko client id
2020-05-11 11:04:40 - robottelo.ssh - INFO - Connected to [FQDN]
2020-05-11 11:04:40 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-05-11 11:04:42 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-7.el7sat.noarch

2020-05-11 11:04:42 - robottelo.ssh - DEBUG - Destroyed Paramiko client id
2020-05-11 11:04:42 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-05-11 11:04:42 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 2 items

tests/foreman/api/test_discoveryrule.py 2020-05-11 11:04:42 - robottelo - INFO - Started setUpClass: tests.foreman.api.test_discoveryrule/DiscoveryRuleTe
2020-05-11 11:04:42 - robottelo - DEBUG - Started Test: DiscoveryRuleTestCase/test_negative_create_with_invalid_host_limit_and_priority
2020-05-11 11:04:42 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/hostgroups with options {'auth': ('', ''), 'verify': False
2020-05-11 11:04:44 - nailgun.client - DEBUG - Received HTTP 201 response: {"subnet_id":null,"subnet_name":null,"operatingsystem_id":null,"operatingsyste
2020-05-11 11:04:44 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/hostgroups/268 with options {'auth': ('', ''), 'verify': Fa
2020-05-11 11:04:45 - nailgun.client - DEBUG - Received HTTP 200 response: {"content_source_id":null,"content_source_name":null,"content_view_id":null,"c
2020-05-11 11:04:45 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/discovery_rules with options {'auth': ('', ''), 'verify': 
2020-05-11 11:04:46 - nailgun.client - WARNING - Received HTTP 422 response: {
  "error": {"id":null,"errors":{"max_count":["is not a number"]},"full_messages":["Max count is not a number"]}
}

2020-05-11 11:04:46 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/hostgroups with options {'auth': ('', ''), 'verify': False
2020-05-11 11:04:47 - nailgun.client - DEBUG - Received HTTP 201 response: {"subnet_id":null,"subnet_name":null,"operatingsystem_id":null,"operatingsyste
2020-05-11 11:04:47 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/hostgroups/269 with options {'auth': ('', ''), 'verify': Fa
2020-05-11 11:04:48 - nailgun.client - DEBUG - Received HTTP 200 response: {"content_source_id":null,"content_source_name":null,"content_view_id":null,"c
2020-05-11 11:04:48 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/discovery_rules with options {'auth': ('', ''), 'verify': 
2020-05-11 11:04:49 - nailgun.client - WARNING - Received HTTP 422 response: {
  "error": {"id":null,"errors":{"priority":["is not a number"]},"full_messages":["Priority is not a number"]}
}

.2020-05-11 11:04:49 - robottelo - DEBUG - Finished Test: DiscoveryRuleTestCase/test_negative_create_with_invalid_host_limit_and_priority
2020-05-11 11:04:49 - robottelo - DEBUG - Started Test: DiscoveryRuleTestCase/test_positive_end_to_end
2020-05-11 11:04:49 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/katello/api/v2/organizations with options {'auth': ('', ''), 'ver
2020-05-11 11:04:57 - nailgun.client - DEBUG - Received HTTP 200 response:   {"label":"giASHdotm","owner_details":{"created":"2020-05-11T09:04:54+0000","

2020-05-11 11:04:57 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/katello/api/v2/organizations/366 with options {'auth': ('', ''), '
2020-05-11 11:04:58 - nailgun.client - DEBUG - Received HTTP 200 response: {"label":"giASHdotm","owner_details":{"created":"2020-05-11T09:04:54+0000","up
2020-05-11 11:04:58 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/locations with options {'auth': ('', ''), 'verify': False,
2020-05-11 11:05:01 - nailgun.client - DEBUG - Received HTTP 201 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"c
2020-05-11 11:05:01 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/locations/367 with options {'auth': ('', ''), 'verify': Fal
2020-05-11 11:05:03 - nailgun.client - DEBUG - Received HTTP 200 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"c
2020-05-11 11:05:03 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/hostgroups with options {'auth': ('', ''), 'verify': False
2020-05-11 11:05:04 - nailgun.client - DEBUG - Received HTTP 201 response: {"subnet_id":null,"subnet_name":null,"operatingsystem_id":null,"operatingsyste
2020-05-11 11:05:04 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/hostgroups/270 with options {'auth': ('', ''), 'verify': Fa
2020-05-11 11:05:05 - nailgun.client - DEBUG - Received HTTP 200 response: {"content_source_id":null,"content_source_name":null,"content_view_id":null,"c
2020-05-11 11:05:05 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/discovery_rules with options {'auth': ('', ''), 'verify': 
2020-05-11 11:05:07 - nailgun.client - DEBUG - Received HTTP 201 response: {"id":193,"name":"RGSA7KMqf382pyEE5j2xtGFGVGKOgEX8hKzuR6Cs6morau7vkikDu83KNNir
2020-05-11 11:05:07 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/discovery_rules/193 with options {'auth': ('', ''), 'verify
2020-05-11 11:05:08 - nailgun.client - DEBUG - Received HTTP 200 response: {"name":"RGSA7KMqf382pyEE5j2xtGFGVGKOgEX8hKzuR6Cs6morau7vkikDu83KNNirY4LwVBXgn
2020-05-11 11:05:08 - nailgun.client - DEBUG - Making HTTP PUT request to https://FQDN/api/v2/discovery_rules/193 with options {'auth': ('', ''), 'verify
2020-05-11 11:05:09 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":193,"name":"RGSA7KMqf382pyEE5j2xtGFGVGKOgEX8hKzuR6Cs6morau7vkikDu83KNNir
2020-05-11 11:05:09 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/katello/api/v2/organizations/366 with options {'auth': ('', ''), '
2020-05-11 11:05:12 - nailgun.client - DEBUG - Received HTTP 200 response: {"label":"giASHdotm","owner_details":{"created":"2020-05-11T09:04:54+0000","up
2020-05-11 11:05:12 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/locations/367 with options {'auth': ('', ''), 'verify': Fal
2020-05-11 11:05:13 - nailgun.client - DEBUG - Received HTTP 200 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"c
2020-05-11 11:05:13 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/katello/api/v2/organizations with options {'auth': ('', ''), 'ver
2020-05-11 11:05:21 - nailgun.client - DEBUG - Received HTTP 200 response:   {"label":"PQFxyPi","owner_details":{"created":"2020-05-11T09:05:17+0000","up

2020-05-11 11:05:21 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/katello/api/v2/organizations/368 with options {'auth': ('', ''), '
2020-05-11 11:05:22 - nailgun.client - DEBUG - Received HTTP 200 response: {"label":"PQFxyPi","owner_details":{"created":"2020-05-11T09:05:17+0000","upda
2020-05-11 11:05:22 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/locations with options {'auth': ('', ''), 'verify': False,
2020-05-11 11:05:25 - nailgun.client - DEBUG - Received HTTP 201 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"c
2020-05-11 11:05:25 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/locations/369 with options {'auth': ('', ''), 'verify': Fal
2020-05-11 11:05:27 - nailgun.client - DEBUG - Received HTTP 200 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"c
2020-05-11 11:05:27 - nailgun.client - DEBUG - Making HTTP POST request to https://FQDN/api/v2/hostgroups with options {'auth': ('', ''), 'verify': False
2020-05-11 11:05:28 - nailgun.client - DEBUG - Received HTTP 201 response: {"subnet_id":null,"subnet_name":null,"operatingsystem_id":null,"operatingsyste
2020-05-11 11:05:28 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/hostgroups/271 with options {'auth': ('', ''), 'verify': Fa
2020-05-11 11:05:29 - nailgun.client - DEBUG - Received HTTP 200 response: {"content_source_id":null,"content_source_name":null,"content_view_id":null,"c
2020-05-11 11:05:29 - nailgun.client - DEBUG - Making HTTP PUT request to https://FQDN/api/v2/discovery_rules/193 with options {'auth': ('', ''), 'verify
2020-05-11 11:05:30 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":193,"name":"ÊäóÚÍáæÒðÐÜÁÓÆÜÏÍÏåÆÎìðùÉÅÀõÀæÙÌáôÎìÞçÍõÞÔØåßÆÐÜëùÆþøêç","se
2020-05-11 11:05:30 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/discovery_rules/193 with options {'auth': ('', ''), 'verify
2020-05-11 11:05:31 - nailgun.client - DEBUG - Received HTTP 200 response: {"name":"ÊäóÚÍáæÒðÐÜÁÓÆÜÏÍÏåÆÎìðùÉÅÀõÀæÙÌáôÎìÞçÍõÞÔØåßÆÐÜëùÆþøêç","enabled":fa
2020-05-11 11:05:31 - nailgun.client - DEBUG - Making HTTP PUT request to https://FQDN/api/v2/discovery_rules/193 with options {'auth': ('', ''), 'verify
2020-05-11 11:05:32 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":193,"name":"ÊäóÚÍáæÒðÐÜÁÓÆÜÏÍÏåÆÎìðùÉÅÀõÀæÙÌáôÎìÞçÍõÞÔØåßÆÐÜëùÆþøêç","se
2020-05-11 11:05:32 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/katello/api/v2/organizations/368 with options {'auth': ('', ''), '
2020-05-11 11:05:34 - nailgun.client - DEBUG - Received HTTP 200 response: {"label":"PQFxyPi","owner_details":{"created":"2020-05-11T09:05:17+0000","upda
2020-05-11 11:05:34 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/locations/369 with options {'auth': ('', ''), 'verify': Fal
2020-05-11 11:05:36 - nailgun.client - DEBUG - Received HTTP 200 response: {"select_all_types":["ProvisioningTemplate","Hostgroup"],"description":null,"c
2020-05-11 11:05:36 - nailgun.client - DEBUG - Making HTTP DELETE request to https://FQDN/api/v2/discovery_rules/193 with options {'auth': ('', ''), 'ver
2020-05-11 11:05:37 - nailgun.client - DEBUG - Received HTTP 200 response: {"id":193,"name":"ÊäóÚÍáæÒðÐÜÁÓÆÜÏÍÏåÆÎìðùÉÅÀõÀæÙÌáôÎìÞçÍõÞÔØåßÆÐÜëùÆþøêç","se
2020-05-11 11:05:37 - nailgun.client - DEBUG - Making HTTP GET request to https://FQDN/api/v2/discovery_rules/193 with options {'auth': ('', ''), 'verify
2020-05-11 11:05:37 - nailgun.client - WARNING - Received HTTP 404 response: {
  "error": {"message":"Resource discovery_rule not found by id '193'"}
}

.2020-05-11 11:05:37 - robottelo - DEBUG - Finished Test: DiscoveryRuleTestCase/test_positive_end_to_end
2020-05-11 11:05:37 - robottelo - INFO - Started tearDownClass: tests.foreman.api.test_discoveryrule/DiscoveryRuleTestCase


========================== 2 passed in 58.73 seconds ===========================

rplevka commented:
@dvagala No, the logs are now as expected - I think it\s okay for the fixture logs to append before "started test" log - I like them to be separated. I'd just like to see them encapsulated in "started fixture for xyz" and "finished fixture for xyz" but I don't think you can do anything about it here.
I don't want such logs to be manually put inside every single fixture.

ACK for me here

jyejare commented on tests/foreman/api/test_discoveryrule.py:
```diff
@@ -24,91 +24,95 @@
-            entities.DiscoveryRule(priority=gen_string('alpha')).create()
+
+
+@pytest.fixture(scope="module")
```
@dvagala , If you have module level API based fixtures, you should add them here [1] so that all the tests can use them.
[1] : https://github.com/SatelliteQE/robottelo/blob/master/robottelo/api/entity_fixtures.py

jyejare commented on tests/foreman/api/test_discoveryrule.py:
```diff
@@ -24,91 +24,95 @@
+
+
+@tier1
+def test_positive_end_to_end_crud(module_org, module_location, module_hostgroup):
```
Module org is already defined entity_fixtures file commented above.

dvagala replied:
@jyejare  Yes, they are defined in entity_fixure.py, but there's no clean up after creating those entities. Those fixtures I've defined in test_discoveryrule.py are basically a wrapper that uses them. Therefore even if the test DiscoveryRule fails, those org and location entities will be deleted properly. I wasn't sure if I should place them in entity_fixtures.py or modify those existing ones. Didn't want to break smth.

jyejare commented:
Oops! This has been merged! You can consider moving them to the appropriate place in separate PR.
